test/features/parsoid/ondemand.js

```diff
@@ -121,4 +121,30 @@ describe('on-demand generation of html and data-parsoid', function() {
         });
     });
 
+    it('should allow full retrieval', function () {
+        var slice = server.config.logStream.slice();
+        return preq.get({
+            uri: contentUrl + '/html/' + revB,
+        })
+        .then(function (res) {
+            slice.halt();
+            assert.deepEqual(res.status, 200);
+            assert.ok(/^<!DOCTYPE html>[^<]*<html[^>]*>/.test(res.body));
+            assert.ok(/<\/html>$/.test(res.body));
+        });
+    });
+
+    it('should allow body-only retrieval', function () {
+        var slice = server.config.logStream.slice();
+        return preq.get({
+            uri: contentUrl + '/html/' + revB + '?bodyOnly',
```

Comment:
We don't currently have a use case for supporting bodyOnly in GET, and I would argue that using query strings would be a bad idea for caching. Lets perhaps hold off on this, so that we can come up with a single consistent API that covers body-only retrieval along with section retrieval?